src/main/java/com/saralein/server/request/RequestParser.java

```diff
@@ -10,11 +10,15 @@ public RequestParser() {
         this.parsedRequest = new HashMap<>();
     }
 
-    public Request parse(String request) {
-        String[] fullRequest = split(request, CRLF);
-        createRequestMap(fullRequest);
+    public Request parse(String request) throws Exception {
+        try {
+            String[] fullRequest = split(request, CRLF);
+            createRequestMap(fullRequest);
```

Comment:
No change is necessary here, but I am always skeptical of side-effect methods. I tend to find it easier to reason about when the method returns the thing it is operating on, or returns a new version of that object. Just a heads up 😸